client/starwhale/api/_impl/model.py

```diff
@@ -288,36 +304,36 @@ def _starwhale_internal_run_ppl(self) -> None:
                 exception = None
                 self._sw_logger.info(f"[{data.idx}] data handle -> success")
 
-            self._do_record(pred, pr, data, label, exception)
+            self._do_record(data, label, exception, *pred)
 
     def _do_record(
         self,
-        pred: t.Any,
-        pr: t.Any,
         data: DataField,
         label: DataField,
         exception: t.Union[None, Exception],
+        *args,
     ) -> None:
         self._status_writer.write(
             {
                 "time": now_str(),  # type: ignore
                 "status": exception is None,
                 "exception": str(exception),
                 "index": data.idx,
-                "output_size": len(pred),
+                "output_size": len(args[0]),
```

Comment:
if `args` is empty, `args[0]` will raise IndexError ?

Reply:
can we assume that `ppl`  function will always return something for `cmp`?

Reply:
> can we assume that `ppl` function will always return something for `cmp`?

yes.